src/Serializer/SerializerContextBuilder.php

```diff
@@ -64,6 +64,10 @@ public function createFromRequest(Request $request, bool $normalization, array $
         if (!$normalization) {
             if (!isset($context['api_allow_update'])) {
                 $context['api_allow_update'] = \in_array($request->getMethod(), ['PUT', 'PATCH'], true);
+
+                if ($context['api_allow_update'] && 'PATCH' === $request->getMethod()) {
```

Comment:
`$request->getMethod()` is called several times. You should store the result in a variable.